lib/rides/pages/search_suggestion_page.dart

```diff
@@ -248,59 +270,77 @@ class _SearchSuggestionPage extends State<SearchSuggestionPage> {
   Widget buildLocationPicker({bool isStart = true}) {
     TextEditingController controller = isStart ? _startController : _destinationController;
 
-    return Expanded(
-      child: ElevatedButton(
-        onPressed: () async {
-          final AddressSuggestion? suggestion = await showSearch<AddressSuggestion?>(
-            context: context,
-            delegate: AddressSearchDelegate(),
-            query: controller.text,
-          );
-          if (suggestion != null) {
-            controller.text = suggestion.name;
-            if (isStart) {
-              _startSuggestion = suggestion;
-            } else {
-              _destinationSuggestion = suggestion;
-            }
-            loadRides();
+    return ElevatedButton(
+      onPressed: () async {
+        final AddressSuggestion? suggestion = await showSearch<AddressSuggestion?>(
+          context: context,
+          delegate: AddressSearchDelegate(),
+          query: controller.text,
+        );
+        if (suggestion != null) {
+          controller.text = suggestion.name;
+          if (isStart) {
+            _startSuggestion = suggestion;
+          } else {
+            _destinationSuggestion = suggestion;
           }
-        },
-        child: Align(
-          alignment: Alignment.centerLeft,
-          child: Text(controller.text),
-        ),
+          loadRides();
+        }
+      },
+      child: Align(
+        alignment: Alignment.centerLeft,
+        child: Text(controller.text),
       ),
     );
   }
 
   Widget buildSearchCardList() {
-    return _rideSuggestions == null
-        ? const Center(child: CircularProgressIndicator())
-        : Expanded(
-            child: ListView.separated(
-              itemBuilder: (context, index) {
-                final ride = _rideSuggestions![index];
-                return SearchCard(ride);
-              },
-              separatorBuilder: (BuildContext context, int index) {
-                return const SizedBox(height: 10);
-              },
-              itemCount: _rideSuggestions!.length,
-            ),
-          );
+    if (_rideSuggestions == null) {
+      return const Center(child: CircularProgressIndicator());
+    }
+    List<Ride> filteredSuggestions = _applyFilters(_rideSuggestions!);
+    Widget list;
+    if (filteredSuggestions.isEmpty) {
+      list = Center(
+        child: SingleChildScrollView(
+          physics: const AlwaysScrollableScrollPhysics(),
+          child: Column(
+            children: [
+              Image.asset('assets/shrug.png'),
+              Text(
+                S.of(context).pageSearchSuggestionsEmpty,
+                style: Theme.of(context).textTheme.displaySmall,
```

Comment:
this is a little bit too big for my taste. Also, this should probably also link to the filter page if filters are set: "Relax your search restrictions..."

Reply:
I didn't include an outright filter button because there is one probably less than 100 pixels above the text. But this is also very useful if you know that if the search results are really empty (e.g. there is no ride found for that route at all), it displays that too, now

Reply:
Well we can just make the text clickable without making it too obvious? It's just something the user might try out...